pkg/apis/operator/v1alpha1/type.go

```diff
@@ -0,0 +1,776 @@
+/*
+Copyright 2022 The Karmada operator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +kubebuilder:path=karmadadeployments,shortName=kmd;kmds,scope=Namespaced
+// +kubebuilder:subresource:status
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:printcolumn:name="Status",type=string,JSONPath=".status.controlPlaneReady",description=""
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp",description="CreationTimestamp is a timestamp representing the server time when this object was created. It is not guaranteed to be set in happens-before order across separate operations. Clients may not set this value. It is represented in RFC3339 form and is in UTC."
+
+// KarmadaDeployment enables declarative installation of karmada.
+type KarmadaDeployment struct {
```

Comment:
Do we have other alternatives for the resource name? 
My concern is the name may be easy to think of as another `application` type from Karmada.

Reply:
how about `karmada` ?

Reply:
How do you think of `KarmadaOperator`, `Karmada` or `Operator`? @RainbowMango

Reply:
/cc @RainbowMango

Reply:
No better ideas yet, I just echo some examples here:
- Istio:  [IstioOperator](https://github.com/istio/istio/blob/18c05bcb96ea1a4f3353fc4563e6c7bbe56a86dc/operator/pkg/apis/istio/v1alpha1/types.go#L51)
- TBD

Reply:
I just checked some examples, now I think we can name it `KarmadaOperator` or `Karmada`.
Given the new API is under the `operator.karmada.io` group, so we can take the shorter one, that is `Karmada`.

For instance:
```yaml
apiVersion: operator.karmada.io/v1alpha1
kind: Karmada
```
So, what do you think? @carlory @calvin0327 @wawa0210 @lonelyCZ @riverzhang

Reply:
+1

Reply:
+1

Reply:
+1